pipelines/toast_ground_sim.py

```diff
@@ -16,16 +16,39 @@
 import traceback
 
 import numpy as np
-from scipy.constants import degree
+import healpy as hp
 
 import toast
 import toast.tod as tt
 import toast.map as tm
-import toast.qarray as qa
 
 
 XAXIS, YAXIS, ZAXIS = np.eye(3)
 
+def extract_detector_parameters(det, schedules):
+    for _, _, fp in schedules:  # loop through focalplanes
+        if det in fp:
+            return fp[det]["bandcenter_ghz"], fp[det]["bandwidth_ghz"], fp[det]["fwhm"]
+    raise RuntimeError("Cannot find detector {} in any focalplane")
+
+def extract_local_dets(data):
+    """Extracts the local detectors from the TOD objects
+
+    Some detectors could only appear in some observations, so we need
+    to loop through all observations and accumulate all detectors in
+    a set
+    """
+    local_dets = set()
+    for obs in data.obs:
+        tod = obs['tod']
+        local_dets.update(tod.local_dets)
+    return local_dets
+
+def assemble_map_on_rank0(comm, local_map, n_components, npix):
+    full_maps_rank0 = np.zeros((n_components, npix), dtype=np.float64) if comm.comm_world.rank == 0 else None
+    comm.comm_world.Reduce(local_map, full_maps_rank0, root=0, op=MPI.SUM)
+    return full_maps_rank0
+
```

Comment:
These last 2 functions belong inside the operator which generates timestreams, such as this:  https://github.com/hpc4cmb/toast/pull/122/files#diff-22d562860acea7384e79d8faca453dc5R197
The extract_local_dets() would be called at the very beginning of the exec() method, before the main loop over detectors.  This timestream operator does not exist in this pull request, but the skeleton was in that other example.